boards/same54-xpro/include/periph_conf.h

```diff
@@ -27,9 +27,37 @@ extern "C" {
 #endif
 
 /**
- * @brief   GCLK reference speed
+ * @brief   Use the external oscillator to source all fast clocks.
+ *          This allows us to use the buck voltage regulator for
+ *          maximum power efficiency, but limits the maximum clock
+ *          frequency to 48 MHz.
  */
-#define CLOCK_CORECLOCK     (120000000U)
+#ifndef USE_XOSC_ONLY
+#define USE_XOSC_ONLY       (0)
+#endif
+
+#ifndef CLOCK_CORECLOCK
+#if USE_XOSC_ONLY
+#define CLOCK_CORECLOCK      (48000000U)    /*  48 MHz */
+#else
+#define CLOCK_CORECLOCK     (120000000U)    /* 120 MHz */
+#endif
+#endif
+
+/**
+ * @name    external Oscillator (XOSC0) configuration
+ * @{
+ */
+#define XOSC0_FREQUENCY     (12000000U)     /* 12 MHz */
+/** @} */
+
+/**
+ * @name    32kHz Oscillator configuration
+ * @{
+ */
+#define EXTERNAL_OSC32_SOURCE                    1
+#define ULTRA_LOW_POWER_INTERNAL_OSC_SOURCE      0
```

Comment:
```suggestion
#ifndef EXTERNAL_OSC32_SOURCE     
#define EXTERNAL_OSC32_SOURCE                    1
#endif
#ifndef ULTRA_LOW_POWER_INTERNAL_OSC_SOURCE  
#define ULTRA_LOW_POWER_INTERNAL_OSC_SOURCE      0
#endif
```

Reply:
Also maybe assert that both cant be 1

Reply:
What would overwrite the board configuration here? CFLAGS?

Reply:
Yes.

Reply:
IMO this is a configuration option, so it makes sense for it to be configurable.

Reply:
Then this should be `CONFIG_ULTRA_LOW_POWER_INTERNAL_OSC_SOURCE` and be exposed through KConfig.